server/src/repositories/job.repository.ts

```diff
@@ -249,24 +251,53 @@ export class JobRepository implements IJobRepository {
   }
 
   private getJobOptions(item: JobItem): JobsOptions | null {
+    let opts = { };
     switch (item.name) {
+      case JobName.NOTIFY_ALBUM_UPDATE:
       case JobName.STORAGE_TEMPLATE_MIGRATION_SINGLE: {
-        return { jobId: item.data.id };
+        opts = { jobId: item.data.id };
+        break;
       }
       case JobName.GENERATE_PERSON_THUMBNAIL: {
-        return { priority: 1 };
+        opts = { priority: 1 };
+        break;
       }
       case JobName.QUEUE_FACIAL_RECOGNITION: {
-        return { jobId: JobName.QUEUE_FACIAL_RECOGNITION };
-      }
-
-      default: {
-        return null;
+        opts = { jobId: JobName.QUEUE_FACIAL_RECOGNITION };
+        break;
       }
     }
+    if ((item?.data as IBaseJob)?.delay) {
```

Comment:
This feels weird, I'm not sure the right TypeScript-y way to do this.

Reply:
Instead of adding `delay` to `IBaseJob` as an optional field , you can make a different `IDelayedJob` that requires it and extend `INotifyAlbumUpdateJob` from it. Then you can move this to the case statement for `JobName.NOTIFY_ALBUM_UPDATE` and hopefully get better typing.

Reply:
Fixed, it looks much better now, thanks!